File: front/src/models/Product.ts
// models/Product.ts
export interface Product {
  id: number
  name: string
  description: string
  price: number
  stockQuantity: number
  category?: {
    id: number
    name: string
  }
  categoryId?: number
  categoryName?: string
  lienImage?: string
  imageUrl?: string
  createdAt?: string
  updatedAt?: string
}

export interface ProductFilters {
  search?: string
  category?: string
  minPrice?: number
  maxPrice?: number
}

File: back/src/main/java/com/example/back/model/Product.java
package com.example.back.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String name;

    private String description;

    @Min(0)
    @Column(nullable = false)
    private double price;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Categorie category;

    @Min(0)
    @Column(nullable = false)
    private int stockQuantity;

    private LocalDateTime createdAt = LocalDateTime.now();

    private String lienImage;
}
